martian-pipestance/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+use anyhow::{Context, Result};
+use serde::de::DeserializeOwned;
+use std::path::Path;
+
+pub mod common;
+pub mod critical_path;
+pub mod final_state;
+pub mod perf;
+
+/// A file associated with a martian pipestance such as _perf, _finalstate etc
+/// that can be deserialized into a concrete type
+pub trait PipestanceFile: DeserializeOwned {
+    fn filename() -> &'static str;
```

Comment:
Since this is `'static` you might as well make this a `const` member instead of a method, unless we think this will need to be dynamically dispatched in certain cases.